src/Widgets/PowerModeList.vala

```diff
@@ -0,0 +1,170 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ * SPDX-FileCopyrightText: 2024 elementary, Inc. (https://elementary.io)
+ */
+
+using Power.Services.DBusInterfaces;
+
+public class Power.Widgets.PowerModeList : Gtk.Box {
+    public static bool successfully_initialized { get; private set; default = true; }
```

Comment:
probably should default to false?

Reply:
It starts with true, until we encounter an error during construction of the object and then its set to false.

Reply:
Seems like this could cause a future race where this property claims the class is initialized before its actually finished/ready

Reply:
Wouldn't the check happen after the class has been constructed and allocated? The function in question is get_proxy_sync which is a blocking code. Or am I misunderstanding something?

Reply:
Hm yeah I get what your'e saying. But if someone refactors it in the future to be async. I dunno. I'm not gonna block you on it, but it doesn't seem good to me to like speculate optimistically about future state instead of making sure that we plan to fail in the correct state